forms-flow-components/src/components/CustomComponents/ImportModal.tsx

```diff
@@ -0,0 +1,331 @@
+    setUploadProgress(0);
+  };
+
+  const closeModal = () => {
+    setSelectedFile(null);
+    setUploadProgress(0);
+    setSelectedLayoutOption(null);
+    setSelectedFlowOption(null);
+    setShowFileItems(false);
+    onClose();
+  };
+
+  const onImport = () => {
+    if (selectedFile) {
+      handleImport(selectedFile, uploadActionType.IMPORT,
+        selectedLayoutVersion?.value, selectedFlowVersion?.value);
+    }
+  };
+
+  useEffect(() => {
+    if (fileItems && !importError && Object.values(fileItems).some(item =>
+      item?.majorVersion != null || item?.minorVersion != null)) {
+      setShowFileItems(true);
+    } else if (processVersion?.majorVersion != null || processVersion?.minorVersion != null) {
+      setShowFileItems(true);
+    } else {
+      setShowFileItems(false);
+    }
+  }, [importError, fileItems]);
+
+  useEffect(() => {
+    if (!importModal) {
+      closeModal();
+    }
+  }, [importModal]);
+
+  useEffect(() => {
+    let isMounted = true;
+
+    if (selectedFile) {
+      handleImport(
+        selectedFile,
+        uploadActionType.VALIDATE,
+        selectedLayoutVersion?.value ?? null,
+        selectedFlowVersion?.value ?? null     
+      );
+
+      let start: number | null = null;
+      const duration = 2000;
+
+      const animateProgress = (timestamp: number) => {
+        if (!start) start = timestamp;
+        const progress = Math.min(((timestamp - start) / duration) * 100, 100);
+
+        if (isMounted) {
+          setUploadProgress(progress);
+          setInprogress(progress < 100);
+        }
+
+        if (progress < 100) {
+          requestAnimationFrame(animateProgress);
+        }
+      };
+
+      const animation = requestAnimationFrame(animateProgress);
+
+      return () => {
+        isMounted = false;
+        cancelAnimationFrame(animation);
+      };
+    }
+  }, [selectedFile]);
+
+  return (
+    <Modal show={importModal} onHide={closeModal} centered size="sm">
+      <Modal.Header>
+        <Modal.Title>
+          <b><Translation>{(t) => t(headerText)}</Translation></b>
+        </Modal.Title>
+        <div className="d-flex align-items-center">
+          <CloseIcon width={16.5} height={16.5} onClick={() => { resetState(); closeModal(); }} />
+        </div>
+      </Modal.Header>
+      <Modal.Body className="p-5">
+        {selectedFile ? (
+          <>
+            <ProgressBar now={uploadProgress} />
+            <div className="upload-body">
+              <div className="upload-details">
+                <p className="upload-file-name">{selectedFile.name}</p>
+                <span className={`${!importLoader && !importError && !inprogress ? 'upload-status-success' :
+                  !importLoader && importError && !inprogress ? 'upload-status-error' : inprogress ? 'upload-status-progress' : ''}`}>
+                  {!importLoader && !importError && !inprogress ? (
+                    <Translation>{(t) => t("Upload Successful")}</Translation>
```

Comment:
please check whether transaltion can work without <translation> tag . using t() , it should work. pls check and if working, remove <translation> tag in entire FILE